src/postprocessing/least_squares.jl

```diff
@@ -1,437 +1,681 @@
-Specialized version of `newtonls` on 6 variables for parametrized orbit determination 
-with respect to ``A_2`` Yarkovsky non-gravitational coefficient. Returns the `niters`-th 
-iteration and the covariance matrix ``\Gamma``. 
-    
-See sections 5.2 and 5.3 of https://doi.org/10.1017/CBO9781139175371. 
-    
-See also [`newtonls`](@ref).
-    
-# Arguments
-
-- `res`: Vector of residuals.
-- `w`: Vector of weights.
-- `x_0`: First guess for the initial conditions.
-- `niters`: Number of iterations.
-"""
-function newtonls_6v(res, w, x0, niters=5)
-    # Have as many residuals as weights
-    @assert length(res) == length(w)
-    # Number of observations
-    nobs = length(res)
-    # Degrees of freedom
-    npar = 6 # length(x0)
-    # Mean square residual
-    Q = chi2(res, w)/nobs
-    # First guess
-    x_new = x0
-    # Iteration
-    for i in 1:niters
-        # Gradient of Q with respect to x_0
-        dQ = TaylorSeries.gradient(Q)(x_new)[1:6]
-        # Hessian of Q with respect to x_0
-        d2Q = TaylorSeries.hessian(Q, x_new)[1:6,1:6]
-        # Newton update rule
-        Δx = - inv(d2Q)*dQ
-        x_new[1:6] = x_new[1:6] + Δx
-        # Normal matrix
-        C = d2Q/(2/nobs) # C = d2Q/(2/m)
-        @show sqrt(((Δx')*(C*Δx))/npar)
-    end
-    # Normal matrix
-    C = TaylorSeries.hessian(Q, x_new)/(2/nobs) # C = d2Q/(2/m)
-    # Covariance matrix
-    Γ = inv(C[1:6,1:6])
-
-    return x_new, Γ
-end
-
-@doc raw"""
-    newtonls_A2(res, w, x0, niters=5)
-
-Specialized version of `newtonls` with the Newton method only over the seventh degree of 
-freedom, i.e., with respect to ``A_2`` Yarkovsky non-gravitational coefficient. Returns the
-`niters`-th iteration, the covariance matrix ``\Gamma`` and the normal matrix ``C``. 
-    
-See sections 5.2 and 5.3 of https://doi.org/10.1017/CBO9781139175371. 
-    
-See also [`newtonls`](@ref).
-    
-# Arguments
-
-- `res`: Vector of residuals.
-- `w`: Vector of weights.
-- `x_0`: First guess for the initial conditions.
-- `niters`: Number of iterations.
-"""
-function newtonls_A2(res, w, x0, niters=5)
-    # Have as many residuals as weights
-    @assert length(res) == length(w)
-    # Number of observations
-    nobs = length(res)
-    # Degrees of freedom
-    npar = length(x0)
-    # Mean square residual
-    Q = chi2(res, w)/nobs
-    # First guess
-    x_new = x0
-    # Iteration
-    for i in 1:niters
-        # Gradient of Q with respect to x_0
-        dQ = TaylorSeries.gradient(Q)(x_new)
-        # Hessian of Q with respect to x_0
-        d2Q = TaylorSeries.hessian(Q, x_new)
-        # Newton update rule
-        Δx = - inv(d2Q)*dQ
-        x_new[7] = x_new[7] + Δx[7]
-        # Normal matrix
-        C = d2Q/(2/nobs) # C = d2Q/(2/m)
-        @show sqrt(((Δx')*(C*Δx))/npar)
-    end
-    # Normal matrix
-    C = TaylorSeries.hessian(Q, x_new)/(2/nobs) # C = d2Q/(2/m)
-    # Covariance matrix
-    Γ = inv(C)
-
-    return x_new, Γ, C
-end
+@doc raw"""
+    OrbitFit{T <: Real}

```

Comment:
If `OrbitFit` represents a least squares fit, then I suggest to rename it to `LeastSquaresFit`. Also, it can be argued that these sub-routines belong conceptually more to orbit determination, so I also suggest to move this file to the `orbit_determination` folder.

Reply:
I agree that `LeastSquaresFit` is a better name, so I've changed all `OrbitFit` to `LeastSquaresFit` and moved the files accordingly.